commit d3069ecc9f57855863141de1b52ef6a202ffe9f9 (refs/changes/89/489/1)
Author: Cody Peter Mello <cody.mello@joyent.com>
Date:   2016-09-07T23:09:36+00:00 (3 years, 1 month ago)
    
    ZAPI-750 Send FWAPI updates with owner_uuid for tag changes

diff --git a/lib/workflows/job-common.js b/lib/workflows/job-common.js
index 75d75cd..0bceb40 100644
--- a/lib/workflows/job-common.js
+++ b/lib/workflows/job-common.js
@@ -1016,21 +1016,23 @@ function updateFwapi(job, cb) {
     job.log.info({ jobParams: jobParams, update: update }, 'update params');
 
     if (Object.keys(update).length === 0 && job.params.task !== 'destroy') {
-        return cb(null, 'No properties affecting FWAPI found: not updating');
+        cb(null, 'No properties affecting FWAPI found: not updating');
+        return;
     }
 
-    update.owner_uuid = jobParams.owner_uuid;
-    update.server_uuid = jobParams.server_uuid;
+    update.owner_uuid = jobParams.owner_uuid || job.params.owner_uuid;
+    update.server_uuid = jobParams.server_uuid || job.params.server_uuid;
     update.type = type;
     update.uuid = jobParams.uuid || jobParams.vm_uuid || job.params.vm_uuid;
 
-    return fwapi.createUpdate(update, function (err, obj) {
+    fwapi.createUpdate(update, function (err, obj) {
         if (err) {
             job.log.warn(err, 'Error sending update to FWAPI');
-            return cb(null, 'Error updating FWAPI');
+            cb(null, 'Error updating FWAPI');
+            return;
         }
 
-        return cb(null, 'Updated FWAPI with update UUID: ' + obj.update_uuid);
+        cb(null, 'Updated FWAPI with update UUID: ' + obj.update_uuid);
     });
 }
 
